lib/Versions/VersionsBackend.php

```diff
@@ -211,4 +267,67 @@ private function getVersionsFolder(int $folderId): Folder {
 			return $trashRoot->newFolder((string)$folderId);
 		}
 	}
+
+	public function setVersionLabel(IVersion $version, string $label): void {
+		$versionEntity = $this->groupVersionsMapper->findVersionForFileId(
+			$version->getSourceFile()->getId(),
+			$version->getTimestamp(),
+		);
+		if (trim($label) === '') {
+			$label = null;
+		}
+		$versionEntity->setLabel($label ?? '');
+		$this->groupVersionsMapper->update($versionEntity);
+	}
+
+	public function deleteVersion(IVersion $version): void {
+		$sourceFile = $version->getSourceFile();
+		$mount = $sourceFile->getMountPoint();
+
+		if (!($mount instanceof GroupMountPoint)) {
+			return;
+		}
+
+		$versionsFolder = $this->getVersionsFolder($mount->getFolderId())->get((string)$sourceFile->getId());
+		/** @var Folder $versionsFolder */
+		$versionsFolder->get((string)$version->getTimestamp())->delete();
```

Comment:
```suggestion
		$versionsFolder->get((string)$version->getRevisionId())->delete();
```
?
I know it should match the timestamp but…